### PR TITLE
add coder-login

### LINKED ISSUE #67: create a  coder-login module
A module that can automatically login a user to coder deployment inside a workspace.
We have a nice workaround [here](https://github.com/coder/coder/blob/87ebe6c2c2387b4249ba959616cf8a0b73684cbd/dogfood/main.tf#L215-L221)

```shell
# Automatically authenticate the user if they are not already logged in
if ! coder list >/dev/null 2>&1; then
  set +x; coder login --token=$CODER_USER_TOKEN --url=$CODER_DEPLOYMENT_URL
else
  echo "You are already authenticated with Coder"
fi
```

### PR DESCRIPTION
**Pull request by stirby**
Added coder-login to automatically authenticate user in new workspace.
Closes #67

### CODE REVIEW

**Comment by stirby** [comment]
Closes #67.

File: coder-login/README.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+---
+display_name: Coder Login
+description: Automatically logs the user into Coder on their workspace
+icon: /emojis/1f511.png
```
Comment (matifali) [inline coder-login/README.md:4]: Thoughts on using the Coder logo?

**Review by matifali — APPROVED** [review]
Just for the logo change. Otherwise it's perfect. Thanks for doing this.

File: coder-login/main.tf [hunk 2]
```diff
@@ -23,7 +23,7 @@ resource "coder_script" "coder-login" {
     CODER_DEPLOYMENT_URL : data.coder_workspace.me.access_url
   })
   display_name       = "Coder Login"
-  icon               = "/emojis/1f511.png"
+  icon               = "/favicons/favicon.svg"
```
Comment (matifali) [inline coder-login/main.tf]: nice does this work?

File: coder-login/main.tf [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+data "coder_workspace" "me" {}
+
+resource "coder_script" "coder-login" {
+  agent_id = var.agent_id
+  script = templatefile("${path.module}/run.sh", {
+    CODER_USER_TOKEN : data.coder_workspace.me.owner_session_token,
+    CODER_DEPLOYMENT_URL : data.coder_workspace.me.access_url
+  })
+  display_name       = "Coder Login"
+  icon               = "../.icons/coder-white.svg"
```
Comment (matifali) [inline coder-login/main.tf:26]: This is run on the Coder server, so it cannot access the .icons folder in the modules repo. 
You can use /icon/coder.svg after you upload the icon to 
https://github.com/coder/coder/tree/main/site/static/icon
or use full HTML path to the coder icon

**Review by matifali — APPROVED** [review]
a comment about logo in main.tf

File: coder-login/main.tf (hunk 2 above)
Comment (stirby) [reply coder-login/main.tf]: It turned out to be too dark. I have a link from svgur now.

File: coder-login/main.tf (hunk 3 above)
Comment (stirby) [reply coder-login/main.tf]: It works great now.

**Pull request merged**